src/HDF5.jl

```diff
@@ -1378,26 +1379,43 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = []
+    for (dtype, dsize) in zip(membertype, membersize)
+        if dtype == String
+            push!(row, strip(String(read(io, UInt8, dsize)),'\0'))
```

Comment:
Not sure how universal this \0 packing is — numpy sticks these chars into its fixed length strings.